### PR TITLE
Authentification

### PR DESCRIPTION
**Pull request by hswiergiel**

### CODE REVIEW

**Review by FlorianBartier — DISMISSED** [review]
Ok

**Review by FlorianBartier — DISMISSED** [review]
Ok

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
If you use the `useApi` service, use it everywhere. therwise, you'll woner why some calls to your api aren't authorized :/

**Review by FlorianBartier — DISMISSED** [review]
Ok for me

File: frontend/src/components/DisplayOffers/DisplayOffers.jsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 import Style from "./style";
 import { Context } from "../../contexts/Context";
 
 export default function DisplayOffers() {
   const [offers, setOffers] = useState([]);
   const { searchValue } = useContext(Context);
+  const api = useApi();
   useEffect(() => {
-    axios.get(`${import.meta.env.VITE_BACKEND_URL}/offers`).then(({ data }) => {
+    api.get(`${import.meta.env.VITE_BACKEND_URL}/offers`).then(({ data }) => {
```
Comment (LoicBrassart) [inline frontend/src/components/DisplayOffers/DisplayOffers.jsx]: `api.get("/offers")` is enough (<3 this hook)

File: frontend/src/pages/Offer/Offer.jsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+import useApi from "@services/useApi";
 import Style from "./style";
 
 export default function Offer() {
   const [offer, setOffer] = useState({});
   const { id } = useParams();
+  const api = useApi();
   useEffect(() => {
-    axios
+    api
```
Comment (LoicBrassart) [inline frontend/src/pages/Offer/Offer.jsx:11]: Same

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Plz refactor

File: frontend/src/pages/Offer/Offer.jsx (hunk 2 above)
Comment (hswiergiel) [reply frontend/src/pages/Offer/Offer.jsx:11]: done

File: frontend/src/components/DisplayOffers/DisplayOffers.jsx (hunk 1 above)
Comment (hswiergiel) [reply frontend/src/components/DisplayOffers/DisplayOffers.jsx]: done!

File: frontend/src/pages/Offer/Offer.jsx (hunk 2 above)
Comment (hswiergiel) [reply frontend/src/pages/Offer/Offer.jsx:11]: done!

**Review by FlorianBartier — APPROVED** [review]
Ok

**Review by LoicBrassart — APPROVED** [review]
NmE!

**Pull request merged**